larsim/EventGenerator/MultiPart/MultiPartVertex_module.cc

```diff
@@ -0,0 +1,762 @@
+////////////////////////////////////////////////////////////////////////
+// Class:       MultiPartVertex
+// Module Type: producer
+// File:        MultiPartVertex_module.cc
+//
+// Generated at Tue Dec 13 15:48:59 2016 by Kazuhiro Terao using artmod
+// from cetpkgsupport v1_11_00.
+////////////////////////////////////////////////////////////////////////
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art/Framework/Principal/Run.h"
+#include "art/Framework/Principal/SubRun.h"
+//#include "art/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "cetlib/pow.h"
+#include <memory>
+#include <vector>
+
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Random/RandGauss.h"
+#include "TRandom.h"
+#include "larcore/CoreUtils/ServiceUtil.h"
+#include "larcore/Geometry/Geometry.h"
+#include "larcoreobj/SimpleTypesAndConstants/PhysicalConstants.h"
+#include "larcoreobj/SummaryData/RunData.h"
+#include "nurandom/RandomUtils/NuRandomService.h"
+
+#include "nusimdata/SimulationBase/MCParticle.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+
+#include "TDatabasePDG.h"
+#include "TLorentzVector.h"
+
+struct PartGenParam {
+  std::vector<int> pdg;
+  std::vector<double> mass;
+  std::array<size_t, 2> multi;
+  std::array<double, 2> kerange;
+  bool use_mom;
+  double weight;
+};
+
+class MultiPartVertex;
+
+class MultiPartVertex : public art::EDProducer {
+public:
+  explicit MultiPartVertex(fhicl::ParameterSet const& p);
+  // The destructor generated by the compiler is fine for classes
+  // without bare pointers or other resource use.
+  ~MultiPartVertex();
+
+  // Plugins should not be copied or assigned.
+  MultiPartVertex(MultiPartVertex const&) = delete;
+  MultiPartVertex(MultiPartVertex&&) = delete;
+  MultiPartVertex& operator=(MultiPartVertex const&) = delete;
+  MultiPartVertex& operator=(MultiPartVertex&&) = delete;
+
+  // Required functions.
+  void produce(art::Event& e) override;
+
+  void beginRun(art::Run& run) override;
+
+  void GenPosition(double& x, double& y, double& z);
+  void GenBoost(double& bx, double& by, double& bz);
```

Comment:
Instead of passing in the output variables are arguments, both of these functions should return something:

```c++
TVector3 GenPosition();
TVector3 GenBoost();
```